src/serialization/preprocessor.cpp

```diff
@@ -740,32 +740,30 @@ bool operator!=(char lhs, preprocessor_data::token_desc::TOKEN_TYPE rhs)
 }
 
 /** preprocessor_file constructor. */
-preprocessor_file::preprocessor_file(preprocessor_streambuf& t, const std::string& name, std::size_t symbol_index)
+preprocessor_file::preprocessor_file(preprocessor_streambuf& t, const wml_path& name)
 	: preprocessor(t)
 	, files_()
 	, pos_()
 	, end_()
 	, name_(name)
-	, is_directory_(filesystem::is_directory(name))
+	, is_directory_(filesystem::is_directory(name.get_abolute_path()))
 {
 	if(is_directory_) {
-		filesystem::get_files_in_dir(name, &files_, nullptr,
-			filesystem::ENTIRE_FILE_PATH,
-			filesystem::SKIP_MEDIA_DIR,
-			filesystem::DO_REORDER
-		);
+		name.get_files_in_dir(&files_, nullptr);
 
-		for(const std::string& fname : files_) {
-			std::size_t cpos = fname.rfind(" ");
+		for(const wml_path& fname : files_) {
+			std::size_t space_pos = fname.safe_path().rfind(' ');
 
-			if(cpos != std::string::npos && cpos >= symbol_index) {
-				std::stringstream ss;
-				ss << "Found filename containing whitespace: '" << filesystem::base_name(fname)
-				<< "' in included directory '" << name << "'.\nThe included symbol probably looks similar to '"
-				<< filesystem::directory_name(fname.substr(symbol_index)) << "'";
+			if(space_pos != std::string::npos) {
+				int schraeger_pos = fname.safe_path().rfind('/');
```

Comment:
Is "shraeger" the word for "slash" in your native language? I think it'd be better to just call it "slash"... although I guess it's fairly self-explanatory in this case so maybe it doesn't matter.